components/renku_data_services/project/db.py

```diff
@@ -52,21 +60,21 @@ async def get_projects(
 
             pagination = PaginationResponse(page, per_page, n_total_elements, total_pages)
 
-            # TODO: Filter based on the APIUser
-
             return [p.dump() for p in projects_orm], pagination
 
-    async def get_project(self, user: base_models.APIUser, id: str) -> models.Project:
+    async def get_project(self, user: base_models.APIUser, project_id: str) -> models.Project:
         """Get one project from the database."""
+        authorized = await self.project_authz.has_permission(user=user, project_id=project_id, scope=Scope.READ)
+        if not authorized:
+            raise errors.Unauthorized(message="You do not have the required permissions to access this project.")
```

Comment:
This should be a 404. If you do not have access to a project and the project you should not even know it exists.


```suggestion
        if not authorized:
            raise errors.MissingResourceError(message=f"Project with id '{project_id}' does not exist or you do not have access to it.")